tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1527,7 +1527,12 @@ public:
 
       // flush all buffers to disk by destroying the TBufferMerger
       fOutputFiles.clear();
+      // TODO: Remove the calls to TTree::SetMaxTreeSize after issue #6640 is fixed
+      auto old_maxtreesize{TTree::GetMaxTreeSize()};
+      TTree::SetMaxTreeSize(std::numeric_limits<Long64_t>::max());
       fMerger.reset();
+      TTree::SetMaxTreeSize(old_maxtreesize);
```

Comment:
for consistency we should do the same also for single-thread Snapshot, I think?

Reply:
@vepadulano can you do the same for single-thread Snapshot please?

Reply:
Yes, I'll push a new commit with the latest modifications